From 362f1b02c5b3f4f3fef6276e4c324855e0c04703 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Wed, 11 Oct 2017 23:06:55 +0300
Subject: [PATCH 007/249] Update r5 installation instructions

* Global oppo forums are now dead, list the complete
  guide in our wiki instead of linking to external sites

Change-Id: If5959e6562b772685986764e2aa4748e7482ca25
---
 _data/devices/r5.yml                          |  3 +-
 _data/devices/r7plus.yml                      |  2 +-
 .../recovery_install_fastboot_oppo.md         | 38 ++++++++++++++-----
 3 files changed, 32 insertions(+), 11 deletions(-)

diff --git a/_data/devices/r5.yml b/_data/devices/r5.yml
index 7bf9bb5..9cca56e 100644
--- a/_data/devices/r5.yml
+++ b/_data/devices/r5.yml
@@ -25,7 +25,8 @@ network:
 - {tech: 4G, bands: '' }
 no_oem_unlock_switch: true
 peripherals: ['']
-project_spectrum: "http://community.oppo.com/en/forum.php?mod=viewthread&tid=44621"
+project_spectrum: "https://androidfilehost.com/?fid=962021903579484169"
+project_spectrum_recovery: "https://androidfilehost.com/?fid=889964283620757488"
 ram: 2 GB
 recovery_boot: With the device powered down, hold the Volume Down and Power buttons.
 release: 2014 December
diff --git a/_data/devices/r7plus.yml b/_data/devices/r7plus.yml
index c7f2b9d..b02426f 100644
--- a/_data/devices/r7plus.yml
+++ b/_data/devices/r7plus.yml
@@ -18,7 +18,7 @@ gpu: Adreno 405
 has_recovery_partition: true
 height:
 image: r7plus.png
-install_method: fastboot_oppo
+install_method: fastboot_nexus
 kernel: android_kernel_oppo_msm8939
 maintainers: [maniac103]
 name: R7 Plus
diff --git a/_includes/templates/recovery_install_fastboot_oppo.md b/_includes/templates/recovery_install_fastboot_oppo.md
index ca6355d..98124e7 100644
--- a/_includes/templates/recovery_install_fastboot_oppo.md
+++ b/_includes/templates/recovery_install_fastboot_oppo.md
@@ -4,18 +4,38 @@
 {% include warning.html content="Unlocking the bootloader will erase all data on your device!
 Before proceeding, ensure the data you would like to retain is backed up to your PC and/or your Google account, or equivalent." %}
 
+{% if site.data.devices[page.device].project_spectrum_recovery %}
+1. Download [Project Spectrum recovery]({{ site.data.devices[page.device].project_spectrum_recovery }}).
+{% endif %}
+{% if site.data.devices[page.device].project_spectrum %}
+2. Download [Project Spectrum]({{ site.data.devices[page.device].project_spectrum }}).
+{% endif %}
+{% if site.data.devices[page.device].required_bootloader_link %}
+3. Download unlockable bootloader obtained from [here]({{ site.data.devices[page.device].required_bootloader_link }}).
+{% endif %}
+4. Place the downloaded .zip packages on the root of `/sdcard`:
+    * Using adb: `adb push filename.zip /sdcard/`
+    * You can use any method you are comfortable with. `adb` is universal across all devices, and works both in Android and recovery mode, providing
+        USB debugging is enabled.
+5. If you aren't already in recovery, reboot into recovery:
+    * {{ device.recovery_boot }}
+{% if site.data.devices[page.device].project_spectrum_recovery %}
+6. Flash Project Spectrum recovery zip and reboot to recovery again:
+    * {{ device.recovery_boot }}
+{% endif %}
 {% if site.data.devices[page.device].project_spectrum %}
-1. Flash Project Spectrum as outlined [here]({{ site.data.devices[page.device].project_spectrum }}).
+7. Flash Project Spectrum zip.
 {% endif %}
 {% if site.data.devices[page.device].required_bootloader_link %}
-2. Flash unlockable bootloader obtained from [here]({{ site.data.devices[page.device].required_bootloader_link }}) via stock recovery.
+8. Flash unlockable bootloader zip.
 {% endif %}
+9. Reboot to android.
 
 {% unless site.data.devices[page.device].no_oem_unlock_switch %}
-3. Enable OEM unlock in the Developer options under device Settings, if present.
+10. Enable OEM unlock in the Developer options under device Settings, if present.
 {% endunless %}
-4. Connect the device to your PC via USB.
-5. On the computer, open a command prompt (on Windows) or terminal (on Linux or macOS) window, and type:
+11. Connect the device to your PC via USB.
+12. On the computer, open a command prompt (on Windows) or terminal (on Linux or macOS) window, and type:
 
         adb reboot bootloader
 
@@ -24,12 +44,12 @@ Before proceeding, ensure the data you would like to retain is backed up to your
 
     * {{ site.data.devices[page.device].download_boot }}
     {% endif %}
-6. Once the device is in fastboot mode, verify your PC finds it by typing:
+13. Once the device is in fastboot mode, verify your PC finds it by typing:
 
         fastboot devices
 
     {% include tip.html content="If you see `no permissions fastboot` while on Linux or macOS, try running `fastboot` as root." %}
-7. Now type the following command to unlock the bootloader:
+14. Now type the following command to unlock the bootloader:
 
 {% if site.data.devices[page.device].custom_unlock_cmd %}
         {{ site.data.devices[page.device].custom_unlock_cmd }}
@@ -37,7 +57,7 @@ Before proceeding, ensure the data you would like to retain is backed up to your
         fastboot oem unlock
 {% endif %}
 
-8. If the device doesn't automatically reboot, reboot it. It should now be unlocked.
-9. Since the device resets completely, you will need to re-enable USB debugging to continue.
+15. If the device doesn't automatically reboot, reboot it. It should now be unlocked.
+16. Since the device resets completely, you will need to re-enable USB debugging to continue.
 
 {% include templates/recovery_install_fastboot_generic.md %}
-- 
2.17.1

